api/v1alpha1/rabbitmqsecretengineconfig_webhook.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	vault "github.com/hashicorp/vault/api"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var rabbitmqsecretengineconfiglog = logf.Log.WithName("rabbitmqsecretengineconfig-resource")
+
+func (r *RabbitMQSecretEngineConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-redhatcop-redhat-io-v1alpha1-rabbitmqsecretengineconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=redhatcop.redhat.io,resources=rabbitmqsecretengineconfigs,verbs=update,versions=v1alpha1,name=vrabbitmqsecretengineconfig.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &RabbitMQSecretEngineConfig{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *RabbitMQSecretEngineConfig) ValidateCreate() error {
+	rabbitmqsecretengineconfiglog.Info("validate create", "name", r.Name)
+
+	var ctx context.Context
+
+	vaultClient, err := r.Spec.Authentication.GetVaultClient(ctx, r.Namespace)
+	if err != nil {
+		rabbitmqsecretengineconfiglog.Error(err, "unable to create vault client required for instance validation", "instance", r)
+		return err
+	}
+	mounts, err := vaultClient.Sys().ListMounts()
+	if err != nil {
+		// Operator requires permission to list mounts for validation
+		rabbitmqsecretengineconfiglog.Error(err, "failed to list mounts", "instance", r)
+		return err
+	}
+	if mountExists := checkIfMountExists(mounts, string(r.Spec.Path)); mountExists {
```

Comment:
I am confused by this validation logic. We want the mount to exists..but the logic seems to be erroring out if it exists.
Perhaps you wanted to check if it was already configured?

Reply:
we can skip this validation, up to you.

Reply:
can we rename this RabbitMQEngineConfigVaultObject?

Reply:
Renamed and updated usage in controller